### ISSUE TITLE
Trouble having peers exchange routes dynamically...

### ISSUE THREAD

**Issue by ndo360** [freelan-developers/freelan#206]
Greetings!

I have been toying with creating a network with FreeLAN and have run into an issue that I was hoping someone here could help resolve or shed light on. Below I'll explain what is going on in my network at the present moment.

I start up the initial node. I have Relay mode enabled, and ports forwarded. 

My two friends join using my Public IP and Certificates. Both have their ports forwarded, and relay mode enabled as well.

We all three are using a 9.0.0.X IP address, with X being the variable number between us to avoid conflicts.

After a couple of minutes I end the process of my node. What I expected to happen was for the network to stay active and for both of my friends still communicate. However both my friends lost connection to one another the moment I dropped my node.

Is there something in the config that I am missing as to why the network didn't stay running? Is there a setting I need to change or alter to keep the network active? Any advise on what to do would be a great help.

Cheers!
Noah C. Cline

**Comment by richman1000000** [comment]
1) disable relay modes on all nodes if you want to setup full-mesh
2) add dynamic contact certfificate
3) allow contact requests

**Comment by ndo360** [comment]
Thanks for the replay!

Can you elaborate more on the dynamic contact certificate? Do I add the path to the user certificate to that line, or do I do something else... first time and I am unsure how to do it.

Providing an example would be a great help as well. Thanks again!

**Comment by ndo360** [comment]
@richman1000000 Okay, so I did as you suggested. However it didn't seem to work unfortunately. I'll post the configs for each host below. Please do help me solve this issue.

Node A (Me)
`public_endpoint=0.0.0.0
dynamic_contact_file="testerone.crt"
ipv4_address_prefix_length=192.168.143.1/24
ipv6_address_prefix_length=2aa1::1/8
signature_private_key_file="testerone.key"
signature_certificate_file="testerone.crt"
authority_certificate_file="ca.crt"`

Node B (Friend)
`public_endpoint=0.0.0.0
dynamic_contact_file="testertwo.crt"
ipv4_address_prefix_length=192.168.143.2/24
ipv6_address_prefix_length=2aa1::2/8
signature_private_key_file="testertwo.key"
signature_certificate_file="testertwo.crt"
authority_certificate_file="ca.crt"`

Node C (Friend)
`public_endpoint=0.0.0.0
dynamic_contact_file="testerthree.crt"
ipv4_address_prefix_length=192.168.143.3/24
ipv6_address_prefix_length=2aa1::3/8
signature_private_key_file="testerthree.key"
signature_certificate_file="testerthree.crt"
authority_certificate_file="ca.crt"`

They all join on me using my Direct IP in the console. Is there anything wrong with the setup? Everything else is their default value within the config. Let me know if you need any further information!

**Comment by richman1000000** [comment]
Node A (Me) (testOne)
~~dynamic_contact_file=testerone.crt~~
dynamic_contact_file="testerthree.crt" 
dynamic_contact_file="testertwo.crt" 

Node B (Friend) (TestTwo)
dynamic_contact_file="testerone.crt" 
dynamic_contact_file="testerthree.crt" 
~~dynamic_contact_file="testertwo.crt"~~

Node C (Friend)
dynamic_contact_file="testerone.crt" 
~~dynamic_contact_file="testerthree.crt"~~
dynamic_contact_file="testertwo.crt"

Nodes should NOT dynamic contact themselfs!!!!
this is traffic loop on node itselft.

**Comment by richman1000000** [comment]
there were freelan server, which had such things.

I think you can write a script, If you need huge scaele (50 users) but currently, freelan doesn't contains automation mechanism

**Comment by ndo360** [comment]
Could you provide an example? Like for a Server running the script, and others connect and learn the routes due to the scripting?

**Comment by guilhermecugler** [comment]
@ndo360 did you succeed in automatically scaling dynamic contacts?

**Comment by ndo360** [comment]
> @ndo360 did you succeed in automatically scaling dynamic contacts?

Nope. I've been mainly waiting for the 3.0 milestone to release, as it seems some 'supernode' capabilities would be possible.

**Comment by guilhermecugler** [comment]
@ndo360  I did an "adaptation", it hurts to see but for my case it will work.
I created a counter on the host that returns a number when receiving a request, so I take that number and directly modify the client's freelan.cfg file, which will continue with this ip "forever", only supports 255 "registered" clients but it is an amount well above what I need.
